sui/src/bin/validator.rs

```diff
@@ -62,38 +54,30 @@ async fn main() -> Result<(), anyhow::Error> {
 
     let network_config_path = sui_config_dir()?.join(SUI_NETWORK_CONFIG);
 
-    let genesis_conf: GenesisConfig = PersistedConfig::read(&cfg.genesis_config_path)?;
-
-    let authority = if let Some(address) = cfg.address {
-        // Find the network config for this validator
-        genesis_conf
-            .authorities
-            .iter()
-            .find(|x| SuiAddress::from(&x.public_key) == address)
-            .ok_or_else(|| {
-                anyhow!(
-                    "Network configs must include config for address {}",
-                    address
-                )
-            })?
-    } else if let Some(index) = cfg.validator_idx {
-        &genesis_conf.authorities[index]
-    } else {
-        return Err(anyhow!("Must supply either --address of --validator-idx"));
-    };
-
-    let genesis_conf_copy: GenesisConfig = PersistedConfig::read(&cfg.genesis_config_path)?;
-
     let network_config = match (network_config_path.exists(), cfg.force_genesis) {
         (true, false) => PersistedConfig::<NetworkConfig>::read(&network_config_path)?,
 
         // If network.conf is missing, or if --force-genesis is true, we run genesis.
         _ => {
-            let (network_config, _, _) =
-                genesis(genesis_conf_copy, Some(authority.address)).await?;
+            let genesis_conf: GenesisConfig = PersistedConfig::read(&cfg.genesis_config_path)?;
+            let adddress = SuiAddress::from(genesis_conf.key_pair.public_key_bytes());
+            let (network_config, _, _) = genesis(genesis_conf, Some(adddress)).await?;
```

Comment:
Actually, we don't need `address` here
We can infer it from the config
So we should remove the field in genesis method

Reply:
We can get the address from here https://github.com/MystenLabs/sui/blob/1caf0cce8ec7744395b408e7efc2d5ba55b751eb/sui/src/sui_commands.rs#L388

Reply:
right we don't. We should clean that up later, trying to get this PR very small and modest

Reply:
Dont you think its small enough that we can fix it now?

Reply:
I don't think it's tiny enough that we should make it here - calls into `genesis`  is spread into multiple files. Also this PR itself does not trigger `address` becoming unused, so they are sorta orthogonal